Tests/NIOTransportServicesTests/NIOTSConnectionChannelTests.swift

```diff
@@ -102,6 +102,27 @@ final class PromiseOnActiveHandler: ChannelInboundHandler {
     }
 }
 
+@available(OSX 10.14, iOS 12.0, tvOS 12.0, *)
+final class EventWaiter<Event>: ChannelInboundHandler {
+    typealias InboundIn = Any
+    typealias InboundOut = Any
+
+    private var eventWaiter: EventLoopPromise<Event>?
+
+    init(_ eventWaiter: EventLoopPromise<Event>) {
+        self.eventWaiter = eventWaiter
+    }
+
+    func userInboundEventTriggered(context: ChannelHandlerContext, event: Any) {
+        if let event = event as? Event {
+            var promise = Optional<EventLoopPromise<Event>>.none
+            swap(&promise, &self.eventWaiter)
```

Comment:
This doesn't feel very idiomatic, is this working around something?

Reply:
It's an alternative spelling for:

```swift
var promise = self.eventWaiter
self.eventWaiter = nil
promise?.succeed(event)
```

The argument is that we want to succeed this promise exactly once and then nil it out, so we do that.

We don't _have_ to do it here, but it's a habit I've gotten in to over time.

Reply:
Fair enough, it just looked like a curious spelling to me!